tests/test_numpy_array.py

```diff
@@ -583,7 +588,7 @@ def test_argument_conversions(forcecast, contiguity, noconvert):
 def test_dtype_refcount_leak():
     from sys import getrefcount
 
-    dtype = np.dtype(np.float_)
+    dtype = np.dtype(np.float64)
```

Comment:
This removes a test (and adds another one).

Could we keep the existing test?

Reply:
No, it doesn't remove a test and add another one, it just adapts to changed names.

Reply:
> No, it doesn't remove a test and add another one, it just adapts to changed names.

Sorry, that's an unreasonable thing to say. — I know it's pedantic, but see https://github.com/pybind/pybind11/pull/5050#issuecomment-2012747730.

There could well be user code out there that uses `np.float_`. That code was covered here before this PR, but not anymore after this change.

Why do you feel you need to take a shortcut?

* It's not difficult to test both, to be safe.

* If you drop `np.float_` because there is a problem: something is getting swept under the rug here.

Reply:
> something is getting swept under the rug here. 

Of course there are breaking changes in the NumPy Python API here (and with the `"L"` change)!  I don't dispute that and am happy to discuss it.  Maybe NumPy 2 will turn out way too optimistic (I totally think that the Python churn is higher than I like, because I think the subtle changes are likely painful enough).

However, this change doesn't matter for `pybind11`.  Even if we open a NumPy issue and revert it due to this discussion: it would be OK to just change the test anyway as the change is harmless.

Reply:
To be clear: `np.float_` is an alias for `np.float64` which was deleted.

Reply:
Aha :-)

I think that's worth a one-line comment, which might help others when they adapt their code for NumPy 2 (I agree it's not a pybind11 problem, but nice to be helpful, if easy). E.g.

```
# NumPy 2 compatibility: Changed from `np.float_` to `np.float64` (`np.float_` was an alias in NumPy 1, dropped in NumPy 2).
```

More generally:

We should tag all changes here, to address breaking changes in the transition to NumPy 2, with a similar comment.

Then from the pybind11 changelog point to those tags.

Reply:
Is there much of a point?  This will hit users when they update NumPy, not pybind11, and then they should look at: https://numpy.org/devdocs/numpy_2_0_migration_guide.html

Reply:
All I'm suggesting is a few one-line comments that are likely to be helpful across a large user base.

And useful in the future to quickly understand what all the breaking changes were, in the context of the pybind11 unit tests.

I'm failing to see why there is pushback to such a small request to be helpful.

Reply:
Sorry, I don't really follow that it is helpful, but yes, happy to add it.